src/App.tsx

```diff
@@ -15,48 +17,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  name = 'name',
+  length = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sort: string,
+  reverse: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sort !== null) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sort) {
+        case SortType.name:
+          return good1.localeCompare(good2);
+
+        case SortType.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setisReversed] = useState(false);
```

Comment:
The naming convention for the 'setisReversed' function is not consistent. It is recommended to use camel case for function names, so 'setIsReversed' would be more appropriate.